middleware/auth.ts

```diff
@@ -0,0 +1,21 @@
+const isAuthenticated = async (): Promise<boolean> => {
+  const sessionToken = sessionStorage.getItem("AUTH_KEY");
+  if (sessionToken == null) {
+    return false;
+  }
+  const response = await $fetch("/api/login", {
+    method: "GET",
+  });
+  if (response.token === sessionToken) {
+    return true;
+  }
+  return false;
+};
```

Comment:
```suggestion
const isAuthenticated = async (): Promise<boolean> => {
  const sessionToken = sessionStorage.getItem("AUTH_KEY");
  
  if (!sessionToken) {
    return false;
  }
  
  const response = await $fetch("/api/login", {
    method: "GET",
  });
  
  return response.token === sessionToken;
};
```
`response.token === sessionToken` return boolean value, so we can do ` return response.token === sessionToken`

Reply:
done